pkg/cloudprovider/aws/suite_test.go

```diff
@@ -479,121 +314,61 @@ var _ = Describe("Allocation", func() {
+			provisioner.SetDefaults(ctx)
+			constraints, err := v1alpha1.NewConstraints(&provisioner.Spec.Constraints)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(aws.StringValue(constraints.CapacityType)).To(Equal(ec2.DefaultTargetCapacityTypeOnDemand))
 		})
 	})
 	Context("Validation", func() {
 		Context("Cluster", func() {
 			It("should fail if fields are empty", func() {
-				for _, cluster := range []v1alpha3.Cluster{
-					{Endpoint: "https://test-cluster", CABundle: ptr.String("dGVzdC1jbHVzdGVyCg==")},
-					{Name: ptr.String("test-cluster"), CABundle: ptr.String("dGVzdC1jbHVzdGVyCg==")},
-					{CABundle: ptr.String("dGVzdC1jbHVzdGVyCg==")},
-					{Name: ptr.String("test-cluster")},
+				for _, cluster := range []v1alpha1.Cluster{
+					{Endpoint: "https://test-cluster"},
+					{Name: "test-cluster"},
+					{},
+				} {
+					provisioner = ProvisionerWithProvider(provisioner, &v1alpha1.AWS{Cluster: cluster})
+					Expect(provisioner.Validate(ctx)).ToNot(Succeed())
+				}
+			})
+			It("should fail for invalid endpoint", func() {
+				for _, endpoint := range []string{
+					"http",
+					"http:",
+					"http://",
+					"https",
+					"https:",
+					"https://",
+					"I am a meat popsicle",
+					"$(echo foo)",
 				} {
-					provisioner.Spec.Cluster = cluster
+					provisioner = ProvisionerWithProvider(provisioner, &v1alpha1.AWS{Cluster: v1alpha1.Cluster{Name: "test-cluster", Endpoint: endpoint}})
 					Expect(provisioner.Validate(ctx)).ToNot(Succeed())
 				}
 			})
 		})
 		Context("Labels", func() {
-			It("should allow unrecognized labels", func() {
-				provisioner.Spec.Labels = map[string]string{"foo": randomdata.SillyName()}
-				Expect(provisioner.Validate(ctx)).To(Succeed())
-			})
-			It("should fail if unrecognized aws labels", func() {
+			It("should not allow labels with the aws label prefix", func() {
 				provisioner.Spec.Labels = map[string]string{"node.k8s.aws/foo": randomdata.SillyName()}
 				Expect(provisioner.Validate(ctx)).ToNot(Succeed())
 			})
-			It("should support launch templates", func() {
-				provisioner.Spec.Labels = map[string]string{
-					"node.k8s.aws/launch-template-name": randomdata.SillyName(),
-				}
-				Expect(provisioner.Validate(ctx)).To(Succeed())
-			})
-			It("should allow launch template id to be specified alone", func() {
-				provisioner.Spec.Labels = map[string]string{"node.k8s.aws/launch-template-name": randomdata.SillyName()}
-				Expect(provisioner.Validate(ctx)).To(Succeed())
-			})
-			It("should support on demand capacity type", func() {
-				provisioner.Spec.Labels = map[string]string{"node.k8s.aws/capacity-type": CapacityTypeOnDemand}
-				Expect(provisioner.Validate(ctx)).To(Succeed())
-			})
-			It("should support spot capacity type", func() {
-				provisioner.Spec.Labels = map[string]string{"node.k8s.aws/capacity-type": CapacityTypeSpot}
-				Expect(provisioner.Validate(ctx)).To(Succeed())
-			})
-			It("should fail for unrecognized capacity type", func() {
```

Comment:
Just checking - is the equivalent of this test (and some of the others near it) still around somewhere (I might have missed it)? Or is it not longer needed?

Reply:
Line 245